packages/core/admin/admin/src/content-manager/components/DynamicZone/components/Component/util.js

```diff
@@ -0,0 +1,16 @@
+import { useMemo } from 'react';
+import { get, toString } from 'lodash';
+import { useCMEditViewDataManager } from '@strapi/helper-plugin';
+
+function useMainValue(schema, componentFieldName) {
+  const { modifiedData } = useCMEditViewDataManager();
+
+  const mainField = useMemo(() => get(schema, ['settings', 'mainField'], 'id'), [schema]);
+  const displayedValue = toString(
+    get(modifiedData, [...componentFieldName.split('.'), mainField], '')
```

Comment:
```suggestion
    get(modifiedData, [...componentFieldPath, mainField], '')
```

With the previous change, there is no need anymore to concat/ split the name.